contracts/dtoken/src/lib.rs

```diff
@@ -86,6 +86,9 @@ pub struct Contract {
     /// User account_id -> { campaign_id -> reward }
     rewards: HashMap<AccountId, HashMap<String, Reward>>,
 
+    /// campaign_id -> { token_id -> amount}
+    funded_reward_amount: HashMap<String, HashMap<AccountId, Balance>>,
```

Comment:
@dmytro-shash we already have `reward_campaigns` so we can iterate over it and collect total amount of rewards

we either don't need this field 
or make it `Balance` fields that cache amount of funded and unpaid rewards (something like `balance_reserved_for_rewards`)

Reply:
@markblaize we have to prevent situation when users are claiming rewards and overflowing reserved_amount FOR THIS EXACT campaign ( if we have single `balance_reserved_for_rewards` field that means it consists of sum of all  campaign s reward that are running on all markets at this point ) 

so i guess more detailed structure could help us manage it more properly :

- leaning on the signature of `claim_rewards(&mut self, campaign_id: String, amount: WBalance) ` function we can easily check whether it is available amount for exact campaign of exact amount to claim so there is reasonable question:


Чи ми точно будемо впевнені що реварди на кампанії не закінчаться і не виникне ситуації, коли доведеться фандити кампанію ще раз( тоді треба буде писати для цього знову функцію) або будемо обмежуватись ситуацією, коли фандимо досить велику кількість ревардів ( які точно не повинні закінчитись ) та обробити ситуацію повернення  ревардів при виданенні кампанії ?